src/transformers/tokenization_utils_fast.py

```diff
@@ -870,3 +877,53 @@ def train_new_from_iterator(
             kwargs["additional_special_tokens"] = additional_special_tokens
 
         return self.__class__(tokenizer_object=tokenizer, **kwargs)
+
+    def update_post_processor(self):
+        """
+        Updates the underlying post processor with the current `bos_token` and `eos_token`.
+        """
+        if not isinstance(self._tokenizer.post_processor, processors.TemplateProcessing) and not isinstance(
+            self._tokenizer.post_processor, processors.Sequence
+        ):
+            return
+
+        bos = self.bos_token
+        bos_token_id = self.bos_token_id
+        if bos is None and self.add_bos_token:
+            raise ValueError("add_bos_token = True but bos_token = None")
+
+        eos = self.eos_token
+        eos_token_id = self.eos_token_id
+        if eos is None and self.add_eos_token:
+            raise ValueError("add_eos_token = True but eos_token = None")
+
+        single = f"{(bos + ':0 ') if self.add_bos_token else ''}$A:0{(' ' + eos + ':0') if self.add_eos_token else ''}"
+        pair = f"{single}{(' ' + bos + ':1') if self.add_bos_token else ''} $B:1{(' ' + eos + ':1') if self.add_eos_token else ''}"
```

Comment:
```suggestion
        pair = f"{single}{(' ' + bos + ':1') if self.add_bos_token else ''} $B:1{(' ' + eos + ':1') if self.add_eos_token else ''}"
``` 
here you are always adding a prefix space before the bos no?

Reply:
here the pair is set by adding onto the `single` var, so the space is just necessary for the syntax of the template. for example in `DebertaConverter` we have

```python
single="[CLS]:0 $A:0 [SEP]:0",
pair="[CLS]:0 $A:0 [SEP]:0 $B:1 [SEP]:1",
```
which could be written as:

```python
single="[CLS]:0 $A:0 [SEP]:0",
pair=f"{single} $B:1 [SEP]:1", # space before  $B
```

Reply:
Ah right right sorry

Reply:
I had a doubt!